backend/dataall/modules/dataset_sharing/services/share_object_service.py

```diff
@@ -42,12 +44,33 @@
 from dataall.modules.datasets_base.db.dataset_repositories import DatasetRepository
 from dataall.modules.datasets_base.db.dataset_models import DatasetTable, Dataset
 from dataall.modules.datasets_base.services.permissions import DATASET_TABLE_READ
+from dataall.base.aws.iam import IAM
+from dataall.base.aws.sts import SessionHelper
+
 import logging
 
 log = logging.getLogger(__name__)
 
 
 class ShareObjectService:
+    @staticmethod
+    def verify_principal_role(session, share: ShareObject) -> bool:
+        role_name = share.principalIAMRoleName
+        account_id = SessionHelper.get_account()
+        delegation_role = SessionHelper.get_delegation_role_arn(account_id)
+        principal_role = IAM.get_role_arn_by_name(account_id, role_name, delegation_role)
```

Comment:
We do not need to send the delegation_role as it defaults to it

Reply:
adressed in [e6d8](https://github.com/data-dot-all/dataall/pull/1161/commits/127e6d87e4368ce0c8e761f0ac7a66ab6ae04da9)